components/cdc/src/endpoint.rs

```diff
@@ -721,6 +727,24 @@ impl<T: 'static + CdcHandle<E>, E: KvEngine, S: StoreRegionMeta> Endpoint<T, E,
             return;
         }
 
+        let scan_task_count = self.scan_task_count.load(Ordering::Relaxed);
+        if scan_task_count + 1 > self.config.incremental_scan_concurrency_limit as isize {
```

Comment:
Should we take the current memory.usage into consideration as an assistant condition at the same time ? If the given node has insufficient memory, this judgement might be not useful ?

Reply:
It's better to have a fixed upper limit. Memory changes rapidly, e.g., scan tasks take 10% memory may be okay if tikv is in idle, but certainly too much when tikv is busy which exhausts almost all memory.